File: php/app/controller/TaskController.php
<?php
declare(strict_types=1);

namespace app\controller;

use app\exception\MolizhishuApiException;
use app\service\MolizhishuClient;
use app\service\TaskRepository;
use app\service\TaskSyncService;
use think\facade\Config;
use think\facade\Log;
use think\Request;
use think\Response;

class TaskController
{
    public function __construct(
        private readonly MolizhishuClient $client,
        private readonly TaskRepository $tasks,
        private readonly TaskSyncService $syncService
    ) {
    }

    public function create(Request $request): Response
    {
        $started = microtime(true);
        $payload = $request->param();

        $error = $this->validateSubmitPayload($payload);
        if ($error !== null) {
            return json(['success' => false, 'message' => $error], 422);
        }

        if (empty($payload['callbackUrl'])) {
            $callbackUrl = Config::get('molizhishu.callback_url');
            if ($callbackUrl) {
                $payload['callbackUrl'] = $callbackUrl;
            } else {
                Log::warning('[local] source=local-api:submit-task callback_url=null message="task will not receive callback"');
            }
        }

        try {
            $data = $this->client->submitTask($payload);
            $this->tasks->saveSubmittedTask($payload, $data);
            return json(['success' => true, 'data' => $data]);
        } catch (MolizhishuApiException $e) {
            return json([
                'success' => false,
                'code' => $e->businessCode,
                'message' => $e->getMessage(),
            ], $e->httpStatus && $e->httpStatus >= 400 ? $e->httpStatus : 502);
        } finally {
            $duration = (int) round((microtime(true) - $started) * 1000);
            Log::info(sprintf('[local] method=POST path=/api/tasks duration=%dms', $duration));
        }
    }

    public function index(Request $request): Response
    {
        $started = microtime(true);
        $page = max(1, (int) $request->get('page', 1));
        $size = min(100, max(1, (int) $request->get('size', 20)));
        $status = $request->get('status');

        try {
            return json(['success' => true, 'data' => $this->tasks->paginateTasks($page, $size, $status ?: null)]);
        } finally {
            $duration = (int) round((microtime(true) - $started) * 1000);
            Log::info(sprintf('[local] method=GET path=/api/tasks source=local-read duration=%dms', $duration));
        }
    }

    public function read(string $taskId): Response
    {
        $started = microtime(true);
        try {
            $task = $this->tasks->getTaskDetail($taskId);
            if (!$task) {
                return json(['success' => false, 'message' => '任务不存在'], 404);
            }
            return json(['success' => true, 'data' => $task]);
        } finally {
            $duration = (int) round((microtime(true) - $started) * 1000);
            Log::info(sprintf('[local] method=GET path=/api/tasks/%s source=local-read duration=%dms', $taskId, $duration));
        }
    }

    public function sync(string $taskId): Response
    {
        $started = microtime(true);
        try {
            return json(['success' => true, 'data' => $this->syncService->syncOne($taskId)]);
        } catch (MolizhishuApiException $e) {
            return json([
                'success' => false,
                'code' => $e->businessCode,
                'message' => $e->getMessage(),
            ], $e->httpStatus && $e->httpStatus >= 400 ? $e->httpStatus : 502);
        } finally {
            $duration = (int) round((microtime(true) - $started) * 1000);
            Log::info(sprintf('[local] method=POST path=/api/tasks/%s/sync source=local-api:manual-compensation duration=%dms', $taskId, $duration));
        }
    }

    public function stop(string $taskId): Response
    {
        $started = microtime(true);
        try {
            $stopResult = $this->client->stopTask($taskId);
            $syncResult = $this->syncService->syncOne($taskId, 'local-api:stop-task:sync-after-stop');

            return json([
                'success' => true,
                'data' => [
                    'message' => $stopResult,
                    'sync' => $syncResult,
                ],
            ]);
        } catch (MolizhishuApiException $e) {
            return json([
                'success' => false,
                'code' => $e->businessCode,
                'message' => $e->getMessage(),
            ], $e->httpStatus && $e->httpStatus >= 400 ? $e->httpStatus : 502);
        } finally {
            $duration = (int) round((microtime(true) - $started) * 1000);
            Log::info(sprintf('[local] method=PUT path=/api/tasks/%s/stop source=local-api:stop-task duration=%dms', $taskId, $duration));
        }
    }

    private function validateSubmitPayload(array $payload): ?string
    {
        if (empty($payload['prompts']) || !is_array($payload['prompts'])) {
            return 'prompts 必须是非空数组';
        }
        if (count($payload['prompts']) > 50) {
            return 'prompts 最多 50 个';
        }
        if (empty($payload['platforms']) || !is_array($payload['platforms'])) {
            return 'platforms 必须是非空数组';
        }
        foreach ($payload['platforms'] as $platform) {
            if (empty($platform['platform']) || empty($platform['mode'])) {
                return 'platforms 每一项必须包含 platform 和 mode';
            }
        }
        return null;
    }
}

File: java/src/main/java/com/molizhishu/demo/controller/TaskController.java
package com.molizhishu.demo.controller;

import com.molizhishu.demo.client.MolizhishuApiException;
import com.molizhishu.demo.client.MolizhishuClient;
import com.molizhishu.demo.config.MolizhishuProperties;
import com.molizhishu.demo.service.TaskSyncService;
import com.molizhishu.demo.service.TaskService;
import com.molizhishu.demo.support.ApiEnvelope;
import com.molizhishu.demo.support.ApiKeyMasker;
import com.molizhishu.demo.support.SubmitPayloadValidator;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
public class TaskController {

	private final MolizhishuClient client;

	private final TaskService taskService;

	private final TaskSyncService taskSyncService;

	private final MolizhishuProperties properties;

	public TaskController(MolizhishuClient client, TaskService taskService, TaskSyncService taskSyncService, MolizhishuProperties properties) {
		this.client = client;
		this.taskService = taskService;
		this.taskSyncService = taskSyncService;
		this.properties = properties;
	}

	@GetMapping("/api/health")
	public Map<String, Object> health() {
		return ApiEnvelope.ok(Map.of("service", "molizhishu-api-pub-java", "status", "ok"));
	}

	@PostMapping("/api/tasks")
	public ResponseEntity<Map<String, Object>> createTask(@RequestBody Map<String, Object> payload) {
		String validation = SubmitPayloadValidator.validate(payload);
		if (validation != null) {
			return ResponseEntity.unprocessableEntity().body(ApiEnvelope.error(validation, null));
		}

		if (!payload.containsKey("callbackUrl") && properties.callbackUrl() != null && !properties.callbackUrl().isBlank()) {
			payload.put("callbackUrl", properties.callbackUrl());
		}

		Map<String, Object> data = client.submitTask(payload);
		taskService.saveSubmittedTask(payload, data);
		return ResponseEntity.ok(ApiEnvelope.ok(data));
	}

	@GetMapping("/api/tasks")
	public Map<String, Object> listTasks(@RequestParam(defaultValue = "1") int page,
	                                     @RequestParam(defaultValue = "20") int size,
	                                     @RequestParam(required = false) String status) {

		return ApiEnvelope.ok(taskService.listTasks(Math.max(page, 1), Math.min(Math.max(size, 1), 100), status));
	}

	@GetMapping("/api/tasks/{taskId}")
	public ResponseEntity<Map<String, Object>> getTask(@PathVariable String taskId) {
		Map<String, Object> task = taskService.getTask(taskId);

		if (task == null) {
			return ResponseEntity.status(HttpStatus.NOT_FOUND).body(ApiEnvelope.error("任务不存在", null));
		}

		return ResponseEntity.ok(ApiEnvelope.ok(task));
	}

	@PostMapping("/api/tasks/{taskId}/sync")
	public Map<String, Object> syncTask(@PathVariable String taskId) {
		return ApiEnvelope.ok(taskSyncService.syncOne(taskId, "local-api:manual-compensation"));
	}

	@PutMapping("/api/tasks/{taskId}/stop")
	public Map<String, Object> stopTask(@PathVariable String taskId) {
		return ApiEnvelope.ok(Map.of("message", client.stopTask(taskId)));
	}

	@PostMapping("/webhooks/molizhishu")
	public ResponseEntity<Map<String, Object>> callback(@RequestBody Map<String, Object> payload) {
		if (!payload.containsKey("taskId") || !payload.containsKey("status")) {
			return ResponseEntity.badRequest().body(ApiEnvelope.error("taskId 和 status 必填", null));
		}

		boolean duplicate = taskService.saveCallback(payload);
		return ResponseEntity.ok(ApiEnvelope.ok(Map.of("duplicate", duplicate)));
	}

	@GetMapping("/api/callback-url")
	public Map<String, Object> getCallbackUrl() {
		return ApiEnvelope.ok(client.getCallbackUrl());
	}

	@PutMapping("/api/callback-url")
	public Map<String, Object> updateCallbackUrl(@RequestBody Map<String, Object> payload) {
		return ApiEnvelope.ok(client.updateCallbackUrl(payload.get("callbackUrl") == null ? null : payload.get("callbackUrl").toString()));
	}

	@GetMapping("/api/cities")
	public Map<String, Object> cities() {
		return ApiEnvelope.ok(client.getCities());
	}

	@GetMapping("/api/settings")
	public Map<String, Object> settings() {
		return ApiEnvelope.ok(Map.of(
				"apiKey", ApiKeyMasker.mask(properties.token()),
				"security", Map.of("apiKeyUpdateAllowed", properties.allowApiKeyUpdate())
		));
	}

	@PutMapping("/api/settings/api-key")
	public ResponseEntity<Map<String, Object>> updateApiKey() {
		return ResponseEntity.status(HttpStatus.FORBIDDEN).body(ApiEnvelope.error("Java demo 未启用运行时写入配置，请通过服务端环境变量配置 API Key", null));
	}

	@ExceptionHandler(MolizhishuApiException.class)
	public ResponseEntity<Map<String, Object>> apiException(MolizhishuApiException e) {
		int status = e.getHttpStatus() >= 400 ? e.getHttpStatus() : 502;

		return ResponseEntity.status(status).body(ApiEnvelope.error(e.getMessage(), e.getCode()));
	}

}
